administrator/components/com_scheduler/src/Helper/ExecRuleHelper.php

```diff
@@ -93,9 +93,50 @@ private function getFromTask(string $property, $default = null)
      */
     public function nextExec(bool $string = true, bool $basisNow = false)
     {
-        // Exception handling here
-        switch ($this->type) {
+        $executionRules = $this->getFromTask('execution_rules');
+        $type           = $executionRules['rule-type'];
+        switch ($type) {
+            case 'interval-minutes':
+                $now             = Factory::getDate('now', 'UTC');
+                $intervalMinutes = (int) $executionRules['interval-minutes'];
+                $interval        = new \DateInterval('PT' . $intervalMinutes . 'M');
+                $nextExec        = $now->add($interval);
+                $nextExec        = $string ? $nextExec->toSql() : $nextExec;
+                break;
+            case 'interval-hours':
+                $now           = Factory::getDate('now', 'UTC');
+                $intervalHours = $executionRules['interval-hours'];
+                $interval      = new \DateInterval('PT' . $intervalHours . 'H');
+                $nextExec      = $now->add($interval);
+                $nextExec      = $string ? $nextExec->toSql() : $nextExec;
+                break;
+            case 'interval-days':
+                $now                 = Factory::getDate('now', 'UTC');
+                $intervalDays        = $executionRules['interval-days'];
+                $interval            = new \DateInterval('P' . $intervalDays . 'D');
+                $nextExec            = $now->add($interval);
+                $execTime            = $executionRules['exec-time'];
+                list($hour, $minute) = explode(':', $execTime);
+                $nextExec->setTime($hour, $minute);
+                $nextExec = $string ? $nextExec->toSql() : $nextExec;
+                break;
+            case 'interval-months':
+                $now            = Factory::getDate('now', 'UTC');
+                $intervalMonths = $executionRules['interval-months'];
+                $interval       = new \DateInterval('P' . $intervalMonths . 'M');
+                $nextExec       = $now->add($interval);
+                $execDay        = $executionRules['exec-day'];
+                $nextExecYear   = $nextExec->format('Y');
+                $nextExecMonth  = $nextExec->format('n');
+                $nextExec->setDate($nextExecYear, $nextExecMonth, $execDay);
+
+                $execTime            = $executionRules['exec-time'];
+                list($hour, $minute) = explode(':', $execTime);
+                $nextExec->setTime($hour, $minute);
+                $nextExec = $string ? $nextExec->toSql() : $nextExec;
+                break;
             case 'interval':
+                // We keep this for b/c reasons, can be removed in 7.0.0
```

Comment:
```suggestion
                * @deprecated 4.4 will be removed in 7.0
```
Is there a reason to drop this in 7.0.0, and not in 6.0?

Reply:
We have the same error in 5.x, so I used 7 instead of 6.